### PR TITLE
Handles KFServing v1_beta_1 api request in Torchserve

### PR DESCRIPTION
**Pull request by abishekchiffon**
## Description

Adds inference apis to handle kfserving v1_beta_1 inference requests.
Adds kfserving envelope to wrap and unwrap the kfserving requests.
Adds KFServing wrapper to create a torchserve docker image to be used in KFServing predictor.
Adds kfserving request functionalites in the existing image classifier(mnist), text_classifier and HuggingFaceTransformer.
This PR sits on the top of the envelope PR #749 that adds the envelope and this PR should only be merged after #749 is merged.
This PR sits on top of the PR #521 from [htappen](https://github.com/htappen/serve/tree/issue_411)

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## Feature/Issue validation/testing

-Additional units added to ts/torch_handler/unit_tests. 

-logs:
  1. unit test logs:[unit_test_infer_pr_final.log](https://github.com/pytorch/serve/files/5483130/unit_test_infer_pr_final.log)
  2. gradle build logs:[build_gradle_logs_after_merge.log](https://github.com/pytorch/serve/files/5489229/build_gradle_logs_after_merge.log)
3. regression test suite logs: 
[regression_suite_after_merge.log](https://github.com/pytorch/serve/files/5489234/regression_suite_after_merge.log)

## Checklist:

- [x] Have you added tests that prove your fix is effective or that this feature works?
- [x] New and existing unit tests pass locally with these changes?
- [x] Has code been commented, particularly in hard-to-understand areas?
- [x] Have you made corresponding changes to the documentation?

### CODE REVIEW

File: ts/model_loader.py [hunk 1]
```diff
@@ -74,48 +76,71 @@ def load(self, model_name, model_dir, handler, gpu_id, batch_size):
...
-            if module_name.endswith(".py"):
-                module_name = module_name[:-3]
-            module_name = module_name.split("/")[-1]
-            module = importlib.import_module(module_name)
-            # pylint: disable=unused-variable
-        except ImportError as e:
-            module_name = ".{0}".format(handler)
-            module = importlib.import_module(module_name, 'ts.torch_handler')
-            function_name = None
+            module, function_name = self._load_handler_file(handler)
```
Comment (chauhang) [inline ts/model_loader.py:81]: @abishekchiffon Why has the logic for handling multiple files been removed?

File: ts/torch_handler/vision_handler.py [hunk 2]
```diff
@@ -8,14 +8,16 @@
 import torch
 from PIL import Image
 from .base_handler import BaseHandler
+from captum.attr import IntegratedGradients
```
Comment (chauhang) [inline ts/torch_handler/vision_handler.py:11]: @abishekchiffon Why is this dependency needed in the base vision handler?

File: ts/torch_handler/image_classifier.py [hunk 3]
```diff
@@ -4,6 +4,10 @@
 import torch
 import torch.nn.functional as F
 from torchvision import transforms
+from torchvision import models
```
Comment (chauhang) [inline ts/torch_handler/image_classifier.py:7]: @abishekchiffon Why do need this in the base image handler?

Comment (abishekchiffon) [reply ts/torch_handler/image_classifier.py:7]: We don't need it, Geeta. Lines 7, 9, 10 will be removed.

File: ts/torch_handler/vision_handler.py (hunk 2 above)
Comment (abishekchiffon) [reply ts/torch_handler/vision_handler.py:11]: Inference does not require Integrated Gradients. The explain functions were in vision_handler, it is the remnant of the explain. The line will be removed.

File: ts/model_loader.py (hunk 1 above)
Comment (abishekchiffon) [reply ts/model_loader.py:81]: The logic has not been removed. It has been moved to the function _load_handler_file of the same TsModelLoader class.

File: kubernetes/kfserving/kf_predictor_docker/config_kf.properties [hunk 4]
```diff
@@ -0,0 +1,9 @@
+#Sample config.properties. In production config.properties at /mnt/models/config/config.properties will be used
+inference_address=http://0.0.0.0:8085
+management_address=http://0.0.0.0:8081
```
Comment (maaquib) [inline kubernetes/kfserving/kf_predictor_docker/config_kf.properties:3]: NIT: Is there any reason for not exposing `metrics_address`?

File: kubernetes/kfserving/mnist_readme.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+ curl -H "Content-Type: application/json" --data @kubernetes/kfserving/kf_request_json/mnist.json http://127.0.0.1:8085/v1/models/mnist:predict
+```
+
+The Prediction response is as below :
+
+```bash
+{
+	"predictions" : [
+
+	2
```
Comment (maaquib) [inline kubernetes/kfserving/mnist_readme.md:26]: NIT: Formatting

File: kubernetes/kfserving/image_transformer/image_transformer/transformer_model_repository.py [hunk 6]
```diff
@@ -0,0 +1,35 @@
+import os
+
+import logging
+import tornado.ioloop
```
Comment (maaquib) [inline kubernetes/kfserving/image_transformer/image_transformer/transformer_model_repository.py:4]: NIT: Unused imports

File: examples/image_classifier/mnist/mnist_handler.py [hunk 7]
```diff
@@ -4,6 +4,8 @@
 from torchvision import transforms
 
 from ts.torch_handler.image_classifier import ImageClassifier
+import base64
```
Comment (maaquib) [inline examples/image_classifier/mnist/mnist_handler.py:7]: NIT: Unused imports

File: kubernetes/kfserving/kfserving_wrapper/TorchserveModel.py [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+                                 available.
+
+            tornado.web.HTTPError: If there is a bad response from the http client.
+
+        Returns:
+            Dict: The Response from the input from the inference endpoint.
+        """
+        if not self.predictor_host:
+            raise NotImplementedError
+        logging.info("kfmodel predict request is %s", json.dumps(request))
```
Comment (maaquib) [inline kubernetes/kfserving/kfserving_wrapper/TorchserveModel.py:69]: Given the request could be fairly large, should this be a debug level log?

**Review by maaquib — CHANGES_REQUESTED** [review]
LGTM. Some minor comments inline

File: kubernetes/kfserving/kfserving_wrapper/TorchserveModel.py (hunk 8 above)
Comment (abishekchiffon) [reply kubernetes/kfserving/kfserving_wrapper/TorchserveModel.py]: Yes, Changed it to debug level.

File: examples/image_classifier/mnist/mnist_handler.py (hunk 7 above)
Comment (abishekchiffon) [reply examples/image_classifier/mnist/mnist_handler.py]: The imports are Removed.

File: kubernetes/kfserving/image_transformer/image_transformer/transformer_model_repository.py (hunk 6 above)
Comment (abishekchiffon) [reply kubernetes/kfserving/image_transformer/image_transformer/transformer_model_repository.py]: The imports are Removed.

File: kubernetes/kfserving/mnist_readme.md (hunk 5 above)
Comment (abishekchiffon) [reply kubernetes/kfserving/mnist_readme.md]: Formatted the readme.

File: kubernetes/kfserving/kf_predictor_docker/config_kf.properties (hunk 4 above)
Comment (abishekchiffon) [reply kubernetes/kfserving/kf_predictor_docker/config_kf.properties:3]: Added the metrics address. The kfserving wrapper does not implement metrics API as the 8082 port is not exposed in kfserver but the metrics API can be accessed inside the cluster.

**Review by chauhang — APPROVED** [review]

**Review by maaquib — APPROVED** [review]

**Pull request merged**